docs/en/observability/monitor-java-app.asciidoc

```diff
@@ -0,0 +1,2005 @@
+parameters of where to send the APM data to. Before starting the Java application,
+let’s get an API key for our APM server running in Elastic Cloud.
+
+When you check your deployment in Elastic Cloud and click on `APM` on
+the left, you will see the `APM Server Secret Token`, which you can use.
+Also you can copy the APM endpoint URL from there.
+
+
+[source,bash]
+----
+java -javaagent:/path/to/elastic-apm-agent-1.17.0.jar\
+  -Delastic.apm.service_name=javalin-app \
+  -Delastic.apm.application_packages=de.spinscale.javalin \
+  -Delastic.apm.server_urls=$APM_ENDPOINT_URL \
+  -Delastic.apm.secret_token=PqWTHGtHZS2i0ZuBol \
+  -jar build/libs/javalin-app-all.jar
+----
+
+You could now go ahead and open up the APM UI and you should see the
+data flowing in.
+
+[discrete]
+==== Automatic attachment
+
+If you do not want to change the startup options of your application,
+the standalone agent allows you to attach to running JVMs on a host.
+
+This requires you to download the standalone jar. You can find the link
+on the
+https://www.elastic.co/guide/en/apm/agent/java/current/setup-attach-cli.html[official
+docs].
+
+To list your locally running java application, you can run
+
+[source,bash]
+----
+java -jar /path/to/apm-agent-attach-1.17.0-standalone.jar --list
+----
+
+As I usually run more than a single java app on my system, I specify the
+application to attach to. Also, make sure, that you have stopped your
+Javalin application with the agent already attached and just start a
+regular Javalin app without the agent configured to attach.
+
+[source,bash]
+----
+java -jar /tmp/apm-agent-attach-1.17.0-standalone.jar --pid 30730 \
+  --config service_name=javalin-app \
+  --config application_packages=de.spinscale.javalin \
+  --config server_urls=$APM_ENDPOINT_URL \
+  --config secret_token=PqWTHGtHZS2i0ZuBol
+----
+
+This above message will return something like this:
+
+[source,text]
+----
+2020-07-10 15:04:48.144  INFO Attaching the Elastic APM agent to 30730
+2020-07-10 15:04:49.649  INFO Done
+----
+
+So now the agent was attached to a running application with a special
+configuration.
+
+While both of the first two possibilities work, you can also use the third
+one: using the APM agent as a direct dependency. This will allow to write
+custom spans and transactions within our application.
+
+[discrete]
+==== Programmatic setup
+
+. Add the java agent dependency.
++
+[source,gradle]
+----
+dependencies {
+  ... 
+  implementation 'co.elastic.apm:apm-agent-attach:1.17.0'
+  ...
+}
+----
++
+. Instrument the application right at the start in our `main()` method.
++
+[source,java]
+----
+public static void main(String[] args) {
+    ElasticApmAttacher.attach();
+    ...
+}
+----
++
+We did not configure any endpoint or API tokens yet. While the
+https://www.elastic.co/guide/en/apm/agent/java/current/setup-attach-api.html#setup-attach-api-configuration[documentation]
+recommends using the `src/main/resources/elasticapm.properties` file, I
```

Comment:
Curious how other writers feel about the use of "I" here. I'm OK with it.

Reply:
In a tutorial, I agree.